### ISSUE TITLE
Update readme to include MoD Infini-Former instructions

### ISSUE THREAD

**Issue by dingo-actual** [dingo-actual/infini-transformer#7]
There are a couple of changes that need to be made to the training loop when using MoD Infini-Former, as well as a minor change to the `.forward()` method of the Module that contains MoD Infini-Former layers.

Need to put example usage in the readme.

**Comment by Blaizzy** [comment]
Can’t wait!

**Comment by dingo-actual** [comment]
Just a heads-up: there's an issue dealing with getting the dimensions to line up for variable-length sequences in `CompressiveMemory` that needs to be addressed. For the moment, I'll write up an example training script and put training instructions in the readme.

The good news is that I made a small change to the implementation's structure for inference that greatly simplifies the training process, so the code you'll see is much less complicated.

Also, I've decided to skip perplexity calculations just to keep things uncluttered (and not introducing any new dependencies).

The example training script and update to the readme should be ready sometime after I finish work this evening (Eastern time, US).
